kubernetes/deployments_kubeclient_blackbox_test.go

```diff
@@ -0,0 +1,452 @@
+			},
+		},
+		{
+			labels: map[string]string{"provider": "fabric8"},
+			data:   map[string]string{},
+		},
+		{
+			labels: map[string]string{"provider": "fabric8"},
+			data: map[string]string{
+				"run": "name: Run\nnamespace my-run\norder: 1", // Missing colon
+			},
+			shouldFail: true,
+		},
+		{
+			labels: map[string]string{"provider": "fabric8"},
+			data: map[string]string{
+				"run": "name: Run\nns: my-run\norder: 1", // Missing namespace
+			},
+			shouldFail: true,
+		},
+		{
+			shouldFail: true, // No provider
+		},
+	}
+	kubeGetter := &testKubeGetter{}
+	metricsGetter := &testMetricsGetter{}
+	userNamespace := "myNamespace"
+	config := &kubernetes.KubeClientConfig{
+		ClusterURL:        "http://api.myCluster",
+		BearerToken:       "myToken",
+		UserNamespace:     userNamespace,
+		KubeRESTAPIGetter: kubeGetter,
+		MetricsGetter:     metricsGetter,
+	}
+
+	expectedName := "fabric8-environments"
+	for _, testCase := range testCases {
+		kubeGetter.cmInput = testCase
+		_, err := kubernetes.NewKubeClient(config)
+		if testCase.shouldFail {
+			assert.Error(t, err)
+		} else {
+			if !assert.NoError(t, err) {
+				continue
+			}
+			configMapHolder := kubeGetter.result.configMapHolder
+			if !assert.NotNil(t, configMapHolder, "No ConfigMap created by test") {
+				continue
+			}
+			assert.Equal(t, userNamespace, configMapHolder.namespace, "ConfigMap obtained from wrong namespace")
+			configMap := configMapHolder.configMap
+			if !assert.NotNil(t, configMap, "Never sent ConfigMap GET") {
+				continue
+			}
+			assert.Equal(t, expectedName, configMap.Name, "Incorrect ConfigMap name")
+		}
+	}
+}
+
+func TestGetEnvironment(t *testing.T) {
+	testCases := []*resourceQuotaInput{
+		{
+			name:      "run",
+			namespace: "my-run",
+			hard: map[v1.ResourceName]float64{
+				v1.ResourceLimitsCPU:    0.7,
+				v1.ResourceLimitsMemory: 1024,
+			},
+			used: map[v1.ResourceName]float64{
+				v1.ResourceLimitsCPU:    0.4,
+				v1.ResourceLimitsMemory: 512,
+			},
+		},
+		{
+			name:      "doesNotExist", // Bad environment name
+			namespace: "my-run",
+			hard: map[v1.ResourceName]float64{
+				v1.ResourceLimitsCPU:    0.7,
+				v1.ResourceLimitsMemory: 1024,
+			},
+			used: map[v1.ResourceName]float64{
+				v1.ResourceLimitsCPU:    0.4,
+				v1.ResourceLimitsMemory: 512,
+			},
+			shouldFail: true,
+		},
+		{
+			name:       "run",
+			namespace:  "my-run",
+			shouldFail: true, // No quantities, so our test impl returns nil
+		},
+	}
+	kubeGetter := &testKubeGetter{}
+	metricsGetter := &testMetricsGetter{}
+	config := &kubernetes.KubeClientConfig{
+		ClusterURL:        "http://api.myCluster",
+		BearerToken:       "myToken",
+		UserNamespace:     "myNamespace",
+		KubeRESTAPIGetter: kubeGetter,
+		MetricsGetter:     metricsGetter,
+	}
+
+	kc, err := kubernetes.NewKubeClient(config)
+	assert.NoError(t, err)
+	for _, testCase := range testCases {
+		kubeGetter.rqInput = testCase
```

Comment:
as explained above, you could use sub-tests for the sake of report clarity.

Reply:
I was going with table-driven tests, (https://github.com/golang/go/wiki/TableDrivenTests) since that seems to be popular in Go. I'm fine with using a separate test per input though, this is what I'm used to in the Java world anyway.